src/game.cpp

```diff
@@ -9948,10 +9948,20 @@ void game::wield( item_location &loc )
         add_msg( m_info, "%s", ret.c_str() );
     }
 
-    // Can't use loc.obtain() here because that would cause things to spill
-    u.mod_moves( -loc.obtain_cost( u ) );
-    u.wield( *loc.get_item() );
-    loc.remove_item();
+    // Can't use loc.obtain() here because that would cause things to spill.
+    // The item_location gets invalidated if wielding an item from the inventory due to updating of
+    // the cache, so we have to use u.i_rem() instead to avoid a debug message.
+    const item *target = loc.get_item();
+    item to_wield = *target;
```

Comment:
It seems like there should be a better way to do this but I wasn't thinking of one